astropy/units/quantity.py

```diff
@@ -1452,35 +1467,64 @@ def to_string(self, unit=None, precision=None, format=None, subfmt=None):
         }
         formats["latex_inline"] = formats["latex"]
 
-        if format not in formats:
-            raise ValueError(f"Unknown format '{format}'")
-        elif format is None:
-            if precision is None:
+        format_spec = formatter if isinstance(formatter, str) else None
+
+        if format is None:
+            # format_spec overwrites precision
+            if formatter is None and precision is None:
```

Comment:
I'm not really worried about speed, but for clarity it might help to put this case up before `formats` is defined:
```
        if format is None and formatter is None and precision is None:
            # Fast path for default formatting settings.
            return f"{self.value}{self._unitstr:s}"
```
But up to you.